### PR TITLE
Escape space when passing commands to arguments

### PR DESCRIPTION
**Pull request by djc**
Both of the fixes proposed in #88 don't work for me:

```diff
diff --git a/src/lib.rs b/src/lib.rs
index a83fd94..dde4adb 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -129,7 +129,7 @@ mod arch {
         match workload {
             Workload::Command(c) => {
                 command.arg("-c");
-                command.args(&c);
+                command.arg(&c.join(" "));
             }
             Workload::Pid(p) => {
                 command.arg("-p");
```

or 

```diff

diff --git a/src/lib.rs b/src/lib.rs
index dde4adb..4acd0b2 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -127,7 +127,12 @@ mod arch {
         command.arg("cargo-flamegraph.stacks");
 
         match workload {
-            Workload::Command(c) => {
+            Workload::Command(mut c) => {
+                // dtrace does not preserve quoting in the string passed via -c, so to pass along
+                // multi-word arguments we need to escape any whitespace within each arg.
+                for arg in c[1..].iter_mut() {
+                    *arg = arg.replace(" ", "\\ ")
+                }
                 command.arg("-c");
                 command.arg(&c.join(" "));
             }
```

So I came up with a new version that at least seems to work for me. @vlthr would you mind checking my work? @spacejam I noticed the work in #88 has been sitting there for a while, but this issue makes flamegraph quite a pain to use on macOS. Please consider reviewing and releasing this (or something like it).

(Also I'm happy to help out with maintaining this project if you prefer.)

### CODE REVIEW

**Comment by vlthr** [comment]
This works great for me! (macOS 10.15.4 Catalina)

**Comment by alexhexabeam** [comment]
This works for me as well, on macOS 10.14.6 Mojave.

**Comment by trevyn** [comment]
Also confirming that this fixes the arg-passing issue for me on 10.14.6.

May it save you a few seconds until this is merged:

`cargo install --git "https://github.com/djc/flamegraph" --branch dtrace-args`

File: src/lib.rs [hunk 1]
```diff
@@ -137,8 +132,17 @@ mod arch {
...
+                    if i > 0 {
+                        escaped.push(' ');
+                    }
+                    escaped
+                        .push_str(&arg.replace(" ", "\\ "));
+                }
+
                 command.arg("-c");
-                command.args(&c);
+                command.arg(&escaped);
```
Comment (michaelkirk) [inline src/lib.rs:145]:
```suggestion
                command.arg(c.join(" "));
```

I think we can omit all of the above and pass all of c as a single arg.

File: src/bin/flamegraph.rs [hunk 2]
```diff
@@ -75,14 +79,15 @@ fn main() {
     let flamegraph_filename: PathBuf = opt
         .output
         .take()
-        .unwrap_or("flamegraph.svg".into());
+        .unwrap_or_else(|| "flamegraph.svg".into());
```
Comment (michaelkirk) [inline src/bin/flamegraph.rs:82]: why this change?

File: src/bin/cargo-flamegraph.rs [hunk 3]
```diff
@@ -309,18 +317,22 @@ fn main() {
     build(&opt);
 
     let workload = workload(&opt);
+    if opt.verbose {
```
Comment (michaelkirk) [inline src/bin/cargo-flamegraph.rs:320]: At least one of the dependencies `inferno`, already relies on `log`, maybe better to use that instead of introducing our own log level machinery.

**Review by michaelkirk — COMMENTED** [review]
(Just a drive by reviewer - I have no authority here.)

This fix worked for me as is, though I've suggested a change for a smaller diff.

And not sure about the unrelated changes, but I'll leave that to the maintainer(s). 

Thanks for pushing up this fix @djc!

File: src/bin/flamegraph.rs (hunk 2 above)
Comment (djc) [reply src/bin/flamegraph.rs:82]: Note that this is in a separate commit, with the commit message noting it cleans up some clippy warnings. (clippy warns against the before pattern because it performs a potentially expensive operation regardless of whether it's needed or not.)

**Comment by kdy1** [comment]
I merged this and #102 in my fork.

See: https://github.com/kdy1/flamegraph

**Comment by kevinkassimo** [comment]
Bump this PR again for maintainer attention. This problem is quite annoying.

**Comment by djc** [comment]
(I sent the maintainer an email last week to ask if I could help out with maintaining this crate.)

**Comment by kdy1** [comment]
I created an alternative tool which can do `cargo flamegraph` can do.
See: https://github.com/kdy1/cargo-profile
It also allows profiling cpu usage per function on mac os x.

**Comment by spacejam** [comment]
thanks for your patience folks, I've returned after taking a few months away from the computer to read and spend time with humans :) I'll be cutting a release shortly with the latest batch of updates

bors r+

**Comment by spacejam** [comment]
Version 0.4 has been released which includes these changes!

And welcome @djc, the newest maintainer, to the team :)

**Comment by djc** [comment]
Thanks!
